addons/hr_timesheet/tests/test_portal_timesheet.py

```diff
@@ -68,3 +69,70 @@ def test_action_view_subtask_timesheet(self):
                 self.assertEqual(view_id, form_view_id)
             elif view_type == 'kanban':
                 self.assertEqual(view_id, kanban_view_id)
+
+    def test_timesheet_multi_company_access(self):
+        """ Ensure a user with access to multiple companies can view
+          timesheets linked to sales orders across companies.
+        """
+        # Skip if required modules are not installed
+        if (self.env['ir.module.module'].search([('name', '=', 'sale_management')]).state == 'uninstalled'
```

Comment:
You could move this test into the `sale_timesheet` module, since it already depends on both 
`sale_management` and `hr_timesheet`. That way we’re sure `sale_management` is always installed 
when the test runs.

For the `website` dependency, we can try to reproduce in the test the same effect that `website` 
would normally have (e.g. if `website` changes field *F*, we update that field manually in the test).  
If that’s not feasible, then it’s fine to skip the test when `website` is not installed, but in 
that case, please use:

`self.env['ir.module.module']._get('website')` instead of `search` for the module check.